src/pages/Trace/Instructions/Instructions.tsx

```diff
@@ -1,19 +1,50 @@
 import React, { useRef, useEffect } from 'react';
+import { App } from '../../../canvas/AppController';
 import { NavButton } from 'components/NavButton/NavButton';
-import { InstructionModal } from 'components/InstructionModal/InstructionModal';
+import { Modal } from 'components/Modal/Modal';
 import { PathTypeButtonGrid } from 'components/PathTypeButtonGrid/PathTypeButtonGrid';
+import { PatternPathType, ModalType } from 'canvas/Enums';
+import { ActionButton } from 'components/ActionButton/ActionButton';
 import './Instructions.css';
-import { PatternPathType } from 'canvas/Enums';
+
 
 interface InstructionsProps {
     curPathType: PatternPathType;
     setPathType: React.Dispatch<React.SetStateAction<PatternPathType>>;
 }
 
 export const Instructions: React.FC<InstructionsProps> = ({curPathType, setPathType}) => {
-    const doneButton: NavButton = {label: 'DONE', to: 'AddMeasurement' };
-    const addPathButton: NavButton = {label:'', to: 'AddPath'};
-    const instruction: InstructModal = {text: ['Choose seam, fold, or edge to add to your pattern']};
+    const navButton: Button = {label: 'DONE'};
+    const addPathButton: Button = {label:''};
+    const warning: Modal = {text: ['Your pattern is incomplete. Please add another line.'], type: ModalType.Warning};
+    const instruction: Modal = {text: ['Choose seam, fold, or edge to add to your pattern'], type: ModalType.Instruction};
+    
+    const [showWarning, setShowWarning] = React.useState<boolean>(false);
+    let doneContainer = <></>;
+    const showDoneButton = useRef(App.document.isEmpty());
+    const arePatternsEnclosed = useRef(App.document.arePatternPiecesEnclosed());
+
+    useEffect(() => {
+        showDoneButton.current = App.document.isEmpty();
```

Comment:
I still get confused when reading the code, for example this line: showDoneButton.current =App.document.isEmpty();. It sounds like we want to show the done button when the document is empty. maybe rename to hideDoneButton? or showDoneButton.current = !App.document.isEmpty() ?

Reply:
Thanks for pointing this out. There was a bug in isEmpty. It was returning the opposite of what it should have returned. I fixed it so now showDoneButton.current = !App.document.isEmpty();